dll/win32/shell32/CDefaultContextMenu.cpp

```diff
@@ -124,11 +137,18 @@ class CDefaultContextMenu :
         virtual HRESULT STDMETHODCALLTYPE SetSite(IUnknown *pUnkSite);
         virtual HRESULT STDMETHODCALLTYPE GetSite(REFIID riid, void **ppvSite);
 
+        // IServiceProvider
+        virtual HRESULT STDMETHODCALLTYPE QueryService(REFGUID svc, REFIID riid, void**ppv)
```

Comment:
For future reference, we should implement `IServiceProviderImpl` and related classes in our ATL.